simple_repository_server/routers/simple.py

```diff
@@ -85,20 +106,26 @@ async def project_list(
             media_type=response_format.value,
         )
 
-    @get("/simple/{project_name}/")
+    @get(prefix + "{project_name}/")
     async def simple_project_page(
-        project_name: str,
         request: fastapi.Request,
+        project_name: str,
+        repository: typing.Annotated[SimpleRepository, Depends(repo_factory)],
         response_format: typing.Annotated[content_negotiation.Format, Depends(get_response_format)],
     ) -> Response:
         normed_prj_name = packaging.utils.canonicalize_name(project_name)
         if normed_prj_name != project_name:
+            new_path_params = request.path_params.copy()
+            new_path_params["project_name"] = normed_prj_name
+            correct_url = utils.relative_url_for(
+                request=request,
+                name="simple_project_page",
+                **new_path_params,
+            )
+            if request.url.query:
+                correct_url = correct_url + "?" + request.url.query
```

Comment:
Not obvious if the absence of query strings was a bug, or it's a new feature required for the time machine and how it relates to DI. Perhaps deserves a separate PR.